src/pages/Ecommerce/Ecommerce.jsx

```diff
@@ -1,10 +1,208 @@
+import Cart from './Cart';
+import GalleryComponent from './DesktopGallery/GalleryComponent';
 
 function Ecommerce() {
+  const [nav, showNav] = useState(false);
+  const [toggle, setToggle] = useState(false);
+  const [index] = useState(0);
+  const [items, setItems] = useState(0);
+  let [quantity, setQuantity] = useState('');
+  const [cartModal, setCartModal] = useState(false);
+
+  const openNav = () => {
+    showNav(!nav);
+    toggle ? setToggle(false) : setToggle(true);
+  };
+
+  const increment = () => {
+    quantity++;
+    setQuantity(quantity);
+  };
+
+  const decrement = () => {
+    setQuantity(quantity === 0 ? 0 : --quantity);
+  };
+
+  const handleCartModal = () => {
+    setCartModal(!cartModal);
+  };
+
+  const addToCart = () => {
+    setQuantity(0);
+    setItems(quantity);
+  };
+
+  const handleChange = (e) => {
+    const parsed = parseInt(e.target.value);
+    if (parsed >= 0 && parsed < 1000) {
+      setQuantity(parsed);
+    } else {
+      setQuantity(0);
+    }
+  };
+
   return (
     <div>
       <Header title="Ecommerce" />
+      <div className="ec--body ">
+        <nav className="relative flex justify-between items-center h-[60px]  px-5  xl:border-b xl:mt-10 xl:pb-10 xl:w-[1200px] xl:mx-auto 2xl:w-[1350px] ">
+          <div className="flex items-center justify-between w-[150px] z-[1]    ">
+            <img
+              className="xl:hidden "
+              onClick={openNav}
+              src={hamburger}
+              alt=""
+            />
+            <img className="ml-2" src={logo} alt="" />
+            <div className="hidden xl:block ">
+              <ul className="desktop--navbar ml-5 text-[#69707D] flex justify-between w-[400px] ml-10 text-lg  ">
+                <li>Collections</li>
+                <li>Men</li>
+                <li>Women</li>
+                <li>About</li>
+                <li>Contact</li>
+              </ul>
+            </div>
+          </div>
+          {nav && (
+            <div className="absolute z-[1]  top-0  left-0 w-[70%] h-[950px] sm:w-[50%] sm:h-[1050px] bg-white xl:hidden ">
+              <img
+                className="absolute top-6 left-5"
+                onClick={openNav}
+                src={close}
+              />
+              <ul className="mt-20 ml-5 font-bold h-[220px] flex flex-col justify-around text-xl ">
+                <li>Collections</li>
+                <li>Men</li>
+                <li>Women</li>
+                <li>About</li>
+                <li>Contact</li>
+              </ul>
+            </div>
+          )}
+
+          {toggle ? (
+            <>
+              <div className="absolute top-0 left-0 w-screen h-[950px] sm:h-[1050px] bg-black opacity-90  z-[0] xl:hidden   "></div>
+              <div className="flex items-center z-[1] opacity-50 ">
+                <img className="mr-5" src={cart} alt="" />
+                <img className="w-[30px]" src={avatar} alt="" />
+              </div>
+
+              {items > 0 ? (
```

Comment:
Extract this and whatever else possible and vary the differences according to the value of `toggle`. You can even do non repeating DOM elements.

Reply:
I don't think this will be easy